packages/connection/src/class-utils.php

```diff
@@ -63,14 +64,22 @@ public static function update_user_token( $user_id, $token, $is_master_user ) {
 	}
 
 	/**
-	 * Returns the Jetpack__API_VERSION constant if it exists, else returns a
-	 * default value of 1.
+	 * Filters the value of the api constant.
 	 *
-	 * @return integer
+	 * @param String $constant_value The constant value.
+	 * @param String $constant_name The constant name.
+	 * @return mixed | null
 	 */
-	public static function get_jetpack_api_version() {
-		$api_version = Constants::get_constant( 'JETPACK__API_VERSION' );
-		$api_version = $api_version ? $api_version : self::DEFAULT_JETPACK_API_VERSION;
-		return $api_version;
+	public static function jetpack_api_constant_filter( $constant_value, $constant_name ) {
+		if ( ! is_null( $constant_value ) ) {
+			// If the constant value was already set elsewhere, use that value.
+			return $constant_value;
+		}
+
+		if ( defined( "self::DEFAULT_$constant_name" ) ) {
+			return constant( "self::DEFAULT_$constant_name" );
```

Comment:
This is a very elegant method of providing defaults for every constant out there. Why do we even need setting up and tearing down this filter every time we expect the `get_constant` method to be called? What do you think about assigning that filter once and keeping it there?

Reply:
Updated in 0b39a42.